ngraph_bridge/enable_variable_ops/ngraph_replace_variable_modifiers.cc

```diff
@@ -173,6 +200,142 @@ Status ReplaceModifiers(Graph* graph, int graph_id) {
+      NodeBuilder::NodeOut ndef_accumassign_op =
+          NodeBuilder::NodeOut(accumassign_op, 0);
+
+      bool use_nesterov;
+      TF_RETURN_IF_ERROR(
+          GetNodeAttr(node->attrs(), "use_nesterov", &use_nesterov));
+      Node* ngraphassign_op;
+      if (use_nesterov) {
+        // Equation in this case:
+        //  var.device(d) - = grad * lr() + accum * momentum() * lr();
+
+        string new_name_mul_1 = node->name() + "_Mul1";
+        NodeBuilder::NodeOut ndef_mul_op_1;
+        TF_RETURN_IF_ERROR(CreateBinaryOpNode(
+            &ndef_mul_op_1, new_name_mul_1, "Mul", input_grad, input_lr, dtype,
+            node, graph, false, false));
+
+        string new_name_mul_2 = node->name() + "_Mul2";
+        NodeBuilder::NodeOut ndef_mul_op_2;
+        TF_RETURN_IF_ERROR(CreateBinaryOpNode(
+            &ndef_mul_op_2, new_name_mul_2, "Mul", input_momentum, input_lr,
+            dtype, node, graph, false, false));
+
+        string new_name_mul_3 = node->name() + "_Mul3";
+        NodeBuilder::NodeOut ndef_mul_op_3;
+        TF_RETURN_IF_ERROR(CreateBinaryOpNode(
+            &ndef_mul_op_3, new_name_mul_3, "Mul", ndef_accumassign_op,
+            ndef_mul_op_2, dtype, node, graph, false, false));
+
+        string new_name_add_1 = node->name() + "_Add_1";
+        NodeBuilder::NodeOut ndef_add_op_1;
+        TF_RETURN_IF_ERROR(CreateBinaryOpNode(
+            &ndef_add_op_1, new_name_add_1, "Add", ndef_mul_op_1, ndef_mul_op_3,
+            dtype, node, graph, false, false));
+
+        string new_name_sub = node->name() + "_Sub";
+        NodeBuilder::NodeOut ndef_sub_op;
+        TF_RETURN_IF_ERROR(CreateBinaryOpNode(&ndef_sub_op, new_name_sub, "Sub",
+                                              input_var, ndef_add_op_1, dtype,
+                                              node, graph, false, false));
+
+        string new_name_ngraphassign = node->name() + "_NGraphAssign";
+        TF_RETURN_IF_ERROR(NodeBuilder(new_name_ngraphassign, "NGraphAssign")
+                               .Attr("validate_shape", true)
+                               .Attr("use_locking", true)
+                               .Attr("T", dtype)
+                               .Attr("ngraph_graph_id", 0)
+                               .Input(input_var)
+                               .Input(ndef_sub_op)
+                               .Device(node->assigned_device_name())
+                               .Finalize(graph, &ngraphassign_op));
+        ngraphassign_op->set_assigned_device_name(node->assigned_device_name());
+      }  // use_nesterov
+      else {
+        // Equation in this case:
+        // var.device(d) -= accum * lr();
+
+        string new_name_mul_1 = node->name() + "_Mul1";
+        NodeBuilder::NodeOut ndef_mul_op_1;
+        TF_RETURN_IF_ERROR(CreateBinaryOpNode(
+            &ndef_mul_op_1, new_name_mul_1, "Mul", ndef_accumassign_op,
+            input_lr, dtype, node, graph, false, false));
```

Comment:
Minor: the `accum * lr()` multiplication could be refactored out of the if-else

Reply:
Good point. Will update code.